src/test/ui/coherence/coherence-inherited-assoc-ty-cycle-err.stderr

```diff
@@ -5,11 +5,6 @@ LL | trait Trait<T> { type Assoc; }
    | ^^^^^^^^^^^^^^
    |
    = note: ...which again requires processing `Trait`, completing the cycle
-note: cycle used when coherence checking all impls of trait `Trait`
-  --> $DIR/coherence-inherited-assoc-ty-cycle-err.rs:19:1
-   |
-LL | trait Trait<T> { type Assoc; }
-   | ^^^^^^^^^^^^^^
```

Comment:
Huh. Why does *this* happen...?

Reply:
Well, I think I know — I guess that the new call to `inferred_outlives_of` query is triggering a cycle error, and we did not use the form that supplies a span for the user. We could fix that, I'll leave a comment...

Reply:
so based on the comment [above](https://github.com/rust-lang/rust/pull/53793/files?utf8=%E2%9C%93&diff=unified#r213813922) guess we keep this.

Reply:
sounds right